### PR TITLE
Client part

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/cc @inikulin 
The client scripts and QUnit tests added. @inikulin please, see the structure and naming of the files, maybe we should make some changes.
Now linting is disabled for client scripts and tests because there are a lot of error (more than 6k), we'll fix it but it'll take some time.

Also, I've create a TODO list, what we should do next:
1) Linting (5500 errors in client scripts, 1200 errors in tests)
2) Task templates (run, run_iframe), move any logic from the templates to the scripts
3) Change some QUnit tests with functional tests
4) CI QUnit-testing and functional testing
5) use Hammerhead as a npm module
6) add a postfix to css rules via Hammerhead (move it from Gulpfile.js)
7) DXperience tests
8) Get rid of async

### CODE REVIEW

**Comment by inikulin** [comment]
1)Let's move code from `client/runner` to `client/test-run`. NOTE: there are templates where already, use them.
2)Remove `client/runner/api-v2`
3)In all subpackages, move dependency wrappers to the deps dir, e.g. in `ui` move `testcafe-core.js` and `hammerhead-api.js` to the `deps/testcafe-core.js` and `deps/hammerhead.js` appropriately.

4)In ui move all cursor related files to the `cursor` dir and rename files (remove `cursor-prefix`)

5)Why `ui/core.js` contains just logic related to the `progressPanel`, shouldn't it be in the `progress-panel.js`?

**In addition to the TODO list**:
Get rid of jquery. And fill an issue for each item in the todo list, please.

File: Gulpfile.js [hunk 1]
```diff
@@ -1,23 +1,165 @@
...
+        });
+});
+
+
+//Build
+gulp.task('build', ['build-client-core', 'build-client-runner', 'build-client-ui', 'build-styles'], function () {
     var js = gulp
-        .src('src/**/*.js')
-        .pipe(babel());
+        .src(['src/**/*.js', '!src/client/core/**/*.js', '!src/client/runner/**/*.js', '!src/client/ui/**/*.js'])
```
Comment (inikulin) [inline Gulpfile.js]: I think `!src/client/**/*.js` will be just fine. Make array multiline BTW.

File: src/client/core/const.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export var IFRAME_WATCH_PROPERTY = 'tc_iwp_cf65bc15';
```
Comment (inikulin) [inline src/client/core/const.js]: `export default` as an object

File: src/client/core/cross-domain-messages.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+export var IFRAME_TEST_RUNNER_PING_DISPATCHER_CMD                  = 'pingIFrameDispatcher';
```
Comment (inikulin) [inline src/client/core/cross-domain-messages.js]: `export default` as an object all theses constants

**Comment by inikulin** [comment]
We need a ES6 transition plan as well:
1)Constants (`const` keyword, new naming format)
2)Classes (use where possible, any object which has state now or `init` method or something like this should be turned into the classes and instantiated there necessary)
3)Arrows
4)Promises

More **TODO**:
Get rid of shared constants, use error types from https://github.com/superroma/testcafe-phoenix/blob/master/src/test-error/type.js, test run service message commands from https://github.com/superroma/testcafe-phoenix/blob/master/src/runner/test-run/command.js,
worker commands from https://github.com/superroma/testcafe-phoenix/blob/master/src/browser-connection/command.js

Please, turn all TODOs and ES6 plan into appropriate GH issues.

**Comment by inikulin** [comment]
\r-

**Comment by AlexanderMoskovkin** [comment]
>3)In all subpackages, move dependency wrappers to the deps dir, e.g. in ui move testcafe-core.js and hammerhead-api.js to the deps/testcafe-core.js and deps/hammerhead.js appropriately.
@inikulin `jquery`, `async` are `deps` too?

**Comment by inikulin** [comment]
@AlexanderMoskovkin yes

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by inikulin** [comment]
Looking at this now I think we shouldn't move runner into test-run since they need end up in the different files. Really sorry for this.
\r-

**Comment by AlexanderMoskovkin** [comment]
no problem, I'll rollback this

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
thanks

**Pull request merged**